src/v/cloud_storage/remote.cc

```diff
@@ -792,4 +793,42 @@ cloud_roles::credentials auth_refresh_bg_op::build_static_credentials() const {
       _region_name};
 }
 
+std::vector<s3::object_tag> remote::get_manifest_tags(
+  const model::ntp& ntp, model::initial_revision_id rev) {
+    auto tags = default_partition_manifest_tags;
+    tags.push_back({.key = "rp-ns", .value = ntp.ns()});
+    tags.push_back({.key = "rp-topic", .value = ntp.tp.topic()});
+    tags.push_back(
+      {.key = "rp-part", .value = ssx::sformat("{}", ntp.tp.partition())});
+    tags.push_back({.key = "rp-rev", .value = ssx::sformat("{}", rev)});
+    return tags;
+}
+
+std::vector<s3::object_tag> remote::get_manifest_tags(
+  const model::topic_namespace& tns, model::initial_revision_id rev) {
+    auto tags = default_topic_manifest_tags;
+    tags.push_back({.key = "rp-ns", .value = tns.ns()});
+    tags.push_back({.key = "rp-topic", .value = tns.tp()});
+    tags.push_back({.key = "rp-rev", .value = ssx::sformat("{}", rev)});
+    return tags;
+}
+
+std::vector<s3::object_tag> remote::get_segment_tags(
+  const model::ntp& ntp, model::initial_revision_id rev) {
+    auto tags = default_segment_tags;
+    tags.push_back({.key = "rp-ns", .value = ntp.ns()});
+    tags.push_back({.key = "rp-topic", .value = ntp.tp.topic()});
+    tags.push_back(
+      {.key = "rp-part", .value = ssx::sformat("{}", ntp.tp.partition())});
+    tags.push_back({.key = "rp-rev", .value = ssx::sformat("{}", rev)});
+    return tags;
+}
+
+const std::vector<s3::object_tag> remote::default_segment_tags = {
+  {"rp-type", "segment"}};
+const std::vector<s3::object_tag> remote::default_topic_manifest_tags = {
+  {"rp-type", "topic-manifest"}};
+const std::vector<s3::object_tag> remote::default_partition_manifest_tags = {
+  {"rp-type", "partition-manifest"}};
```

Comment:
there are several strings here, some are repeated. Would it be cleaner to add `constexpr`s for the tag names and the static values?

Reply:
the vector can't be a **constexpr**, the strings are in .data section since they're constants but they still need to be copied per request in any case